android/src/main/java/com/twiliorn/library/CustomTwilioVideoView.java

```diff
@@ -303,30 +308,27 @@ public CustomTwilioVideoView(ThemedReactContext context) {
 
     // ===== SETUP =================================================================================
 
-    private VideoConstraints buildVideoConstraints() {
-        Log.d(TAG,"Setting camera constraints. Max Dimensions: "
-            + this.maxCaptureDimensions + " - Max FPS: " + this.maxCaptureFPS);
-
-        return new VideoConstraints.Builder()
-                .maxVideoDimensions(this.maxCaptureDimensions)
-                .maxFps(this.maxCaptureFPS)
-                .build();
+    private VideoFormat buildVideoFormat() {
+        return new VideoFormat(VideoDimensions.CIF_VIDEO_DIMENSIONS, 15);
```

Comment:
Good spot!

I've restored the usage of `maxCaptureDimensions` & `maxCaptureFPS`.

FYI, as mentioned above, `VideoConstraints` have been replaced by `VideoFormat` in the native Twilio library (hence the change here).

However, it seems we're not currently using these `CameraSettings` on the JS side...

Looking back through the commits, I can [see](https://github.com/senseobservationsystems/goalie-2-mobile-app/blob/698094dab5257e0b4cc382b46edda948e2405886/app/twilio/sagas.ts#L147) that we were passing `maxDimensions` & `maxFPS` in the `setIsVideoEnabled` function prior to my "Call UX improvements" refactor, during which they were removed. 😬

At the time, I assumed `setIsVideoEnabled` wasn't called during initialisation (only when the video button was pressed), and therefore these settings were not being used on the majority of calls. However, having logged it out I can now see that `setIsVideoEnabled` **is** called during initialisation.

I can create a small chore in the goalie repo to put these setting back, but I do wonder if it would be more logical to pass them as part of the `connect` call (like the other parameters) instead. WDYT?

Reply:
Ahh, the impact of not having these params is the video quality on Android phones would be quite low.

Having these params in `Connect` sounds good!

Reply:
Ahh, so they're more about increasing the quality, not decreasing it -- got it!

I've put the removed code back in the [parent PR](https://github.com/senseobservationsystems/goalie-2-mobile-app/pull/4681).

It would be great if you (or another reviewer) could check that the `CameraSettings` are now working as before. I can see that they're passing through to the Android side, but I also get the same video quality with or without them. I'm *guessing* they're only used if the device's hardware supports them (my primary test device an AMG A8)? 

I'll make a follow-up chore to move `CameraSettings` from `setIsVideoEnabled` to `connect` to avoid this PR ballooning.

Reply:
ahh I see there is some confusion here @jamesogrant 

I see you are now passing the `maxDimensions` and `maxFPS` from NiceDay to the SDK but you are not using these parameters in the Java file.

My comment was intended for the line I put this comment on i.e line 312 in CustomTwilioVideoView.java 

In your changes, the function `buildVideoFormat` returns:

`return new VideoFormat(VideoDimensions.CIF_VIDEO_DIMENSIONS, 15);`

which is a hardcoded video format and hence ignores the FPS and dimensions that we passed to the library.

This function should instead be taking in the FPS and Dimensions parameter into consideration as it used to be in the function before this PR i.e:

```java
return new VideoConstraints.Builder()
  .maxVideoDimensions(this.maxCaptureDimensions)
  .maxFps(this.maxCaptureFPS)
  .build();
```

NOTE: The code snippet above was from the previous SDK so you would probably need to change it a bit for it to work with the updated SDK